lib/directory.coffee

```diff
@@ -64,7 +64,7 @@ class Directory
     return unless repo?
 
     @subscriptions.add repo.onDidChangeStatus (event) =>
-      @updateStatus(repo) if event.path.indexOf("#{@path}#{path.sep}") is 0
+      @updateStatus(repo) if event.path.indexOf("#{@realPath}#{path.sep}") is 0
```

Comment:
Should we keep the old condition as well? For files (which are working fine) both `@path` and `@realPath` are checked ([here](https://github.com/atom/tree-view/blob/master/lib/file.coffee#L53) and [here](https://github.com/atom/tree-view/blob/master/lib/file.coffee#L76).)
I haven't seen `repo`'s `onDidChangeStatus` event containing a path with a symlink directory, but I guess that would be a little safer to do.